server/sandbox_run_linux.go

```diff
@@ -962,6 +962,12 @@ func (s *Server) runPodSandbox(ctx context.Context, req *types.RunPodSandboxRequ
 		return nil, ctx.Err()
 	}
 	sb.SetCreated()
+	if s.config.EventedPLEG {
+		if err := s.Runtime().UpdateContainerStatus(ctx, sb.InfraContainer()); err != nil {
```

Comment:
again here. updatecontainerstatus is a fairly expensive call, so we should call it as few times as possible

Reply:
This [link](https://kubernetes.slack.com/archives/C03HA69L6R1/p1658751854843449) actually summarizes the issue and the possible fix looks to be updating the pod status prior to the generation of an event.

cc: @harche